commands/management/add.go

```diff
@@ -16,10 +16,54 @@
 package management
 
 import (
+	"fmt"
+	"net"
+	"strconv"
+
 	"gopkg.in/alecthomas/kingpin.v2"
 )
 
 // register command
 var (
 	AddC = kingpin.Command("add", "Add Service to monitoring")
+	// Add command gobal flags
+	AddAddressFlag     = AddC.Flag("host", "Hostname").String()
+	AddPortFlag        = AddC.Flag("port", "Port number").Int64()
+	AddServiceNameFlag = AddC.Flag("service-name", "Service name").String()
 )
+
+type getter interface {
+	GetAddressPort() string
+	GetServiceName() string
+}
+
+// Types implementing the getter interface:
+// - addMongoDBCommand
+// - addMySQLCommand
+// - addPostgreSQLCommand
+// - addProxySQLCommand
+// Returns, service name, address, port, error
+func processGlobalAddFlags(cmd getter) (string, string, int, error) {
+
+	serviceName := cmd.GetServiceName()
+	if *AddServiceNameFlag != "" {
+		serviceName = *AddServiceNameFlag
+	}
+
+	addressPort := cmd.GetAddressPort()
+	if *AddAddressFlag != "" && *AddPortFlag != 0 {
```

Comment:
AFAIK, each of them should be able to override separately. @AlekSi @rnovikovP?

Reply:
Correct. `--port=12345 1.2.3.4` should set full address to `1.2.3.4:12345`, not `:12345`.